### ISSUE TITLE
The hidden trouble of the ResidualBlock class in your code

### ISSUE THREAD

**Issue by breAchyz** [xueluli/SPG-3DVS#1]
Hi, I'm reproducing your algorithm using `tf2 + keras` and meet a problem that when I trained the network in `mixed_precision` policy, the loss get `Nan`. 

I checked the code and network structure, and I find it's caused by the `ResidualBlock` you used in the code you provide.
The solution I used is adding the `BatchNorm` layer after the `Conv2D` and before the `ReLU` layer in the `ResidualBlock`.
